### DISCUSSION TITLE
Activity function tasks not evenly retrieved & processed by all active & healthy host instances.

### DISCUSSION THREAD

**Discussion by yeefung-hin** [Azure/azure-functions-durable-extension#2991] [Q&A] [answered]
Hi, need comment & advice on one odd scenario that I am currently encountering for my application.

I am running an ADF function app using Microsoft.Azure.WebJobs.Extensions.DurableTask 2.9.0 (we are working on upgrading it) with 4 partitions in dedicated/app service plan with 15 host instances (plan type: Premium v3 P2V3). I noticed that there are always 4 host instances which are running in much higher CPU usage (e.g. 60% vs 10%) than the other 11 host instances.

I am using ApplicationInsights and I have tried to investigate what each host instance are doing via using the following query:-
`requests
| extend requestType = case(
        name contains "Orchestrator", "Orchestrator",
        name contains "Activity", "Activity",
        "HTTP"
    )
| extend hostInstanceId = tostring(customDimensions.HostInstanceId)
| where cloud_RoleName == @CloudRoleName
    and timestamp between (@StartDatetime .. @ToDatetime)
| summarize count() by hostInstanceId, requestType`

**_Note: My application's functions has "Orchestrator" & "Activity" postfix in function name._**

![RequestDistribution](https://github.com/user-attachments/assets/8d3597ce-663e-4199-861c-02ce1a60a15a)
![RequestDistribution_Orchestrator](https://github.com/user-attachments/assets/da9dc164-7756-41ad-ac06-2454d3453a73)
![RequestDistribution_Activity](https://github.com/user-attachments/assets/897f74d5-98ab-45c0-9c0f-ae4994abf966)
![RequestDistribution_Http](https://github.com/user-attachments/assets/df21e403-3bf5-4185-8fee-eb547cb3bf84)

From the chart above, I noticed that the 4 host instances which are handling orchestrator functions are processing activity function tasks way more than the other host instances and the difference is very big (13xxxx vs 6xxx).

I have tried to search for similar discussions and the closest one I found is [issue#804](https://github.com/Azure/azure-functions-durable-extension/issues/804) but the issue is not answered yet. Any suggestions on why or how and where I should start digging for root cause?

Do let me know if you need more information. Thanks.

**Comment by yeefung-hin** [comment]
Hi, I just wanted to check in and see if there’s any update on this Q&A. Let me know if there’s anything I can do to assist. Thanks again for your time and efforts!

**Comment by cgillum** [comment, marked as answer]
In many situations, this can be expected. Orchestrations can only run on workers that own one of the configured partitions. When these orchestrations schedule activities, the worker that scheduled the activity will try to read it from the queue immediately without waiting for the polling interval to minimize queue latency. As a result, workers that own partitions are more likely to execute activities compared to workers that don’t own partitions, especially when load is not high.
